### PR TITLE
Created a new function for an Extracted a PPTX files

### PR DESCRIPTION
**Pull request by Kabilan-16**
created a new function for extract a ppt files

### CODE REVIEW

**Comment by Kabilan-16** [comment]
[OPERATING.pptx](https://github.com/user-attachments/files/18149578/OPERATING.pptx)
[OPERATING.pptx.txt](https://github.com/user-attachments/files/18149603/OPERATING.pptx.txt)
[cyber_security_table.pptx](https://github.com/user-attachments/files/18149605/cyber_security_table.pptx)
[cyber_security_table.pptx.txt](https://github.com/user-attachments/files/18149609/cyber_security_table.pptx.txt)
[Computer_Networks_books.pptx](https://github.com/user-attachments/files/18149610/Computer_Networks_books.pptx)
[Computer_Networks_books.pptx.txt](https://github.com/user-attachments/files/18149611/Computer_Networks_books.pptx.txt)
[cloud_computing_books.pptx](https://github.com/user-attachments/files/18149614/cloud_computing_books.pptx)
[cloud_computing_books.pptx.txt](https://github.com/user-attachments/files/18149616/cloud_computing_books.pptx.txt)

**Review by SaravanakumarR2018 — COMMENTED** [review]
Try checking this file

[Full_Pitch.pptx](https://github.com/user-attachments/files/18151991/Full_Pitch.pptx)

**Comment by Kabilan-16** [comment]
[Full_Pitch.pptx](https://github.com/user-attachments/files/18160282/Full_Pitch.pptx)
[Full_Pitch.pptx.txt](https://github.com/user-attachments/files/18160288/Full_Pitch.pptx.txt)
[OPERATING.pptx](https://github.com/user-attachments/files/18160290/OPERATING.pptx)
[OPERATING.pptx.txt](https://github.com/user-attachments/files/18160294/OPERATING.pptx.txt)
[cyber_security_table.pptx](https://github.com/user-attachments/files/18160299/cyber_security_table.pptx)
[cyber_security_table.pptx.txt](https://github.com/user-attachments/files/18160301/cyber_security_table.pptx.txt)
[Computer_Networks_books.pptx](https://github.com/user-attachments/files/18160305/Computer_Networks_books.pptx)
[Computer_Networks_books.pptx.txt](https://github.com/user-attachments/files/18160307/Computer_Networks_books.pptx.txt)
[cloud_computing_books.pptx](https://github.com/user-attachments/files/18160309/cloud_computing_books.pptx)
[cloud_computing_books.pptx.txt](https://github.com/user-attachments/files/18160311/cloud_computing_books.pptx.txt)

**Review by SaravanakumarR2018 — COMMENTED** [review]
Can you check whether you can retrieve data from the charts like below

image

**Comment by Kabilan-16** [comment]
[Bar_Charts.pptx](https://github.com/user-attachments/files/18177192/Bar_Charts.pptx)
[Bar_Charts.pptx.txt](https://github.com/user-attachments/files/18177193/Bar_Charts.pptx.txt)
[Full_Pitch.pptx](https://github.com/user-attachments/files/18177198/Full_Pitch.pptx)
[Full_Pitch.pptx.txt](https://github.com/user-attachments/files/18177202/Full_Pitch.pptx.txt)
[cloud_computing_books.pptx](https://github.com/user-attachments/files/18177204/cloud_computing_books.pptx)
[cloud_computing_books.pptx.txt](https://github.com/user-attachments/files/18177209/cloud_computing_books.pptx.txt)
[cyber_security_table.pptx](https://github.com/user-attachments/files/18177227/cyber_security_table.pptx)
[cyber_security_table.pptx.txt](https://github.com/user-attachments/files/18177230/cyber_security_table.pptx.txt)
[Computer_Networks_books.pptx](https://github.com/user-attachments/files/18177232/Computer_Networks_books.pptx)
[Computer_Networks_books.pptx.txt](https://github.com/user-attachments/files/18177234/Computer_Networks_books.pptx.txt)
[OPERATING.pptx](https://github.com/user-attachments/files/18177241/OPERATING.pptx)
[OPERATING.pptx.txt](https://github.com/user-attachments/files/18177248/OPERATING.pptx.txt)

File: src/backend/embedding/extract_file/extract_file.py [hunk 1]
```diff
@@ -430,9 +440,97 @@ def extract_docx_content(self, customer_guid: str, file_path: str, filename: str
...
+                        # Append the reconstructed table data
+                        slide_elements.append({
+                            "type": "chart_table",
+                            "content": table_data
+                        })   
+
+                # Combine slide elements
+                slide_text = "\n".join(element["content"] for element in slide_elements)
+                results.append({
+                    "metadata": {"slide_number": slide_number},
```
Comment (SaravanakumarR2018) [inline src/backend/embedding/extract_file/extract_file.py:514]: make it 
 "metadata": {"page_number": slide_number},
 
 to keep it consistent across all document retrievals

File: src/backend/embedding/extract_file/extract_file.py [hunk 2]
```diff
@@ -430,9 +440,97 @@ def extract_docx_content(self, customer_guid: str, file_path: str, filename: str
...
+                            for row in table.rows
+                        ]
+                        table_text = self.format_table_as_text(table_data)
+                        slide_elements.append({
+                            "type": "table",
+                            "content": table_text
+                        })
+
+                    # Extract images
+                    if shape.shape_type == 13:  # 13 indicates a picture
```
Comment (SaravanakumarR2018) [inline src/backend/embedding/extract_file/extract_file.py:477]: Make 13 as an ENUM or make sure you get this value from already declared enum in Presentation library

**Review by SaravanakumarR2018 — COMMENTED** [review]
You have done well
Why some of the texts are not captured like below

image

**Comment by Kabilan-16** [comment]
[Bar_Charts.pptx](https://github.com/user-attachments/files/18180987/Bar_Charts.pptx)
[Bar_Charts.pptx.txt](https://github.com/user-attachments/files/18180991/Bar_Charts.pptx.txt)
[Full_Pitch.pptx](https://github.com/user-attachments/files/18180993/Full_Pitch.pptx)
[Full_Pitch.pptx.txt](https://github.com/user-attachments/files/18180997/Full_Pitch.pptx.txt)
[cloud_computing_books.pptx](https://github.com/user-attachments/files/18181001/cloud_computing_books.pptx)
[cloud_computing_books.pptx.txt](https://github.com/user-attachments/files/18181002/cloud_computing_books.pptx.txt)
[Computer_Networks_books.pptx](https://github.com/user-attachments/files/18181006/Computer_Networks_books.pptx)
[Computer_Networks_books.pptx.txt](https://github.com/user-attachments/files/18181010/Computer_Networks_books.pptx.txt)
[cyber_security_table.pptx](https://github.com/user-attachments/files/18181017/cyber_security_table.pptx)
[cyber_security_table.pptx.txt](https://github.com/user-attachments/files/18181019/cyber_security_table.pptx.txt)
[OPERATING.pptx](https://github.com/user-attachments/files/18181022/OPERATING.pptx)
[OPERATING.pptx.txt](https://github.com/user-attachments/files/18181024/OPERATING.pptx.txt)

**Review by SaravanakumarR2018 — APPROVED** [review]

**Pull request merged**